### PR TITLE
feat: added WebhookCredentials struct for WebhookAuthentication

### PR DESCRIPTION
**Pull request by DTStew**

### CODE REVIEW

File: zendesk/webhook.go [hunk 1]
```diff
@@ -28,10 +28,18 @@ type Webhook struct {
 	UpdatedBy      string                 `json:"updated_by,omitempty"`
 }
 
+type WebhookCredentials struct {
+	HeaderName  string `tfsdk:"name,omitempty"`
```
Comment (JacobPotter) [inline zendesk/webhook.go]: `tfsdk` tag is for terraform, think you want to use `json`

File: zendesk/webhook.go [hunk 2]
```diff
@@ -28,10 +28,18 @@ type Webhook struct {
...
+	Password    string `tfsdk:"password,omitempty"`
+	Token       string `tfsdk:"token,omitempty"`
+}
+
 type WebhookAuthentication struct {
-	Type        string      `json:"type"`
-	Data        interface{} `json:"data"`
-	AddPosition string      `json:"add_position"`
+	Type        string              `json:"type"`
+	Data        *WebhookCredentials `json:"data"`
```
Comment (JacobPotter) [inline zendesk/webhook.go:41]: Why are we using a pointer here?

**Review by JacobPotter — CHANGES_REQUESTED** [review]

File: zendesk/webhook.go (hunk 1 above)
Comment (DTStew) [reply zendesk/webhook.go]: Ah right.  Bad copy paste.

File: zendesk/webhook.go (hunk 2 above)
Comment (DTStew) [reply zendesk/webhook.go:41]: I just saw that's what the author was doing for the other objects, so I copied.  🙂

Comment (JacobPotter) [reply zendesk/webhook.go:41]: Oh yeah I see that now.

**Review by JacobPotter — APPROVED** [review]

**Pull request merged**